feature-odyssey/src/main/kotlin/com/hedvig/android/odyssey/repository/ClaimsFlowRepository.kt

```diff
@@ -0,0 +1,156 @@
+package com.hedvig.android.odyssey.repository
+
+import com.hedvig.android.core.common.await
+import com.hedvig.android.odyssey.model.ClaimState
+import com.hedvig.android.odyssey.model.Input
+import com.hedvig.android.odyssey.model.Resolution
+import com.hedvig.android.odyssey.network.toUpdateRequest
+import com.hedvig.odyssey.remote.money.MonetaryAmount
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import org.json.JSONObject
+
+interface ClaimsFlowRepository {
+  suspend fun getOrCreateClaim(
+    commonClaimId: String?,
+  ): ClaimResult
+
+  suspend fun updateClaim(claimState: ClaimState, nrOfInputs: Int): ClaimResult
+  suspend fun getClaim(): ClaimResult
+  suspend fun openClaim(amount: MonetaryAmount? = null): ClaimResult
+}
+
+sealed interface ClaimResult {
+  data class Success(
+    val claimState: ClaimState,
+    val inputs: List<Input>,
+    val resolution: Resolution,
+  ) : ClaimResult
+
+  data class Error(val message: String) : ClaimResult
+}
+
+class NetworkClaimsFlowRepository(
+  private val okHttpClient: OkHttpClient,
+) : ClaimsFlowRepository {
+
+  private val json = Json {
+    allowSpecialFloatingPointValues = true
+    isLenient = true
+    allowStructuredMapKeys = true
+    ignoreUnknownKeys = true
+  }
```

Comment:
Should probably make one of these and expose it through Koin to everyone who needs it.

Reply:
Good point, we can leave it for now though since this will be replaced by graphql